src/common/s3.connection.cpp

```diff
@@ -23,232 +24,142 @@ common::S3Connection::S3Connection(const std::string& endpoint,
     CHECK(client_);
 }
 
-common::S3Connection::~S3Connection() noexcept
-{
-    client_.reset();
-}
-
 bool
-common::S3Connection::bucket_exists(const std::string& bucket_name)
+common::S3Connection::check_connection()
 {
-    if (bucket_name.empty()) {
-        return false;
-    }
-
-    try {
-        minio::s3::BucketExistsArgs args;
-        args.bucket = bucket_name;
-
-        minio::s3::BucketExistsResponse response = client_->BucketExists(args);
-        CHECK(response);
-
-        return response.exist;
-    } catch (const std::exception& e) {
-        LOGE("Failed to check existence of bucket: %s", e.what());
-    }
-
-    return false;
+    return (bool)client_->ListBuckets();
```

Comment:
This is something I wanted to comment on before. We should prefer `static_cast` to C-style cases because it provides more type safety and makes the intention of the conversion explicit.